geoipupdate/test/test_geoip_update.py

```diff
@@ -1,45 +1,51 @@
 import shutil
 import unittest
 import tempfile
+import os
 
 
 from unittest.mock import patch, MagicMock, mock_open
+from testfixtures import TempDirectory
 
-from ..geoip import match_md5, write, download_file
+from ..geoip import match_sha256, write, download_file
 
 
 class GeoipTestCases(unittest.TestCase):
 
     @patch('urllib.request.urlopen')
-    def test_md5_mismatch(self, mock_urlopen='http://foo'):
-        mock_md5 = MagicMock()
-        mock_md5.getcode.return_value = 200
-        mock_md5.read.return_value = '4059D862F09C315470DC8FF0FF6EBCAF'.encode('utf-8')
-        mock_urlopen.return_value = mock_md5
+    def test_sha256_mismatch(self, mock_urlopen='http://foo'):
+        mock_sha256 = MagicMock()
+        mock_sha256.getcode.return_value = 200
+        mock_sha256.read.return_value = '4059D862F09C315470DC8FF0FF6EBCAF'.encode('utf-8')
```

Comment:
I think you need to change the return value incorporate sha256 hash. I think md5 and sha256 output hash would be different. This doc says one is 32 digits and other is 64 digits. https://www.mobilefish.com/services/hash_generator/hash_generator.php